### DISCUSSION TITLE
Enable resources to have revisions

### DISCUSSION THREAD

**Discussion by edalzell** [statamic-rad-pack/runway#488] [Feature Requests]
### Description

Would be amazing if resources (models) could have revisions, like entries do.

I realize this might require some PRs into Statamic, I'm exploring that.

### Steps to reproduce

1. Try to update a model and have that be a revision

### Environment

N/A

**Comment by edalzell** [comment]
Notes:
* copy most of how the eloquent driver does it
* Create a `HasRevisions` trait for this behaviour
* need a controller to handle the revision stuff, copy most from `EntryRevisionsController`
* figure out where the revisions panel goes

Statamic PR:
* Make Vue component out of the revisions div in `components/entries/PublishForm`
* (this might be a Runway thing) support multiple RevisionRepository classes so that Statamic stuff still works
* (maybe) make the `revision-history` components more generic so we don't have to duplicate them in Runway

**Comment by duncanmcclean** [comment]
Completed in #492.
